src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/AppointmentTimeRequest.kt

```diff
@@ -0,0 +1,14 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model
+
+import com.fasterxml.jackson.annotation.JsonFormat
+import io.swagger.v3.oas.annotations.media.Schema
+import java.time.LocalDateTime
+import javax.validation.constraints.NotNull
+
+@Schema(description = "Request object for updating the date and time of the initial appointment")
+data class AppointmentTimeRequest(
+  @Schema(description = "The date and time of the initial appointment", example = "12/12/2021 10:35")
+  @NotNull
```

Comment:
I think this needs to be `@field:NotNull`. Also, does this throw a `400: Bad request` error from the API if the date is in the wrong format/empty?

https://kotlinlang.org/docs/annotations.html#annotation-use-site-targets

Reply:
@field:NotNull - cant be used on a LocalDateTime unfortunately. I've added controller advice to catch JSON parse exceptions in requests to throw a 400 error + tests to verify.